### ISSUE TITLE
Database dump, not just raw csv file dump

### ISSUE THREAD

**Issue by Solomon** [18F/FEC#16]
Getting FEC data into a database is difficult. https://github.com/Solomon/opensecrets_to_postgres

Since I imagine the data is originally coming from a database, can you open up your nightly database backups?

**Comment by saizai** [comment]
@catherinedevlin The FEC has no authority to place any usage restrictions on the data. It's open government data, no copyright. You can't use donor info for ~ commercial solicitation purposes — but that's not a restriction on dissemination.

**Comment by saizai** [comment]
(Also, I'm looking into hosting a mysql based mirror on makeyourlaws.org.)

**Comment by Solomon** [comment]
@catherinedevlin You might be able to host the file on github releases,
which has no file or bandwidth limit right now (although that might change)
https://help.github.com/articles/about-releases/

**Comment by dwillis** [comment]
@saizai: As your comment makes clear, the FEC has the authority to place usage restrictions on it - donor solicitation is a use - but more to the point, @catherinedevlin isn't making a copyright claim. Making a database dump available isn't a trivial matter, and it needs to be something the FEC can maintain and pay for.

Perhaps more important, you can already load the bulk files. Anyone who could make use of a database dump could also load the files themselves.

**Comment by saizai** [comment]
@dwillis Loading the files is fairly time-consuming etc. Yes, I can do it, but why should everyone have to duplicate the effort vs straight up SQL? One could even have an open SQL master set up so anyone can slave the tables directly…

Anyway, since I'm already loading it in bulk, I'll look into hosting it in SQL format for others too. It's a nice thing for my non-profit to do. ;-)

@Solomon While releases might not have file limits, I github itself does. https://help.github.com/articles/working-with-large-files/

**Comment by dwillis** [comment]
@saizai Yes, loading the files is time-consuming, although you can script it so it happens while you sleep :-) (And you made the initial point about usage restrictions, not me. I was just saying that it was a misplaced criticism, as the FEC clearly has the legal authority to do so.)

I think one good reason why I'd rather the FEC not provide SQL dumps is that it increases the amount of work the FEC has to do (and the work is essentially duplicative). Any time the FEC spends on this is at the expense of something they already are doing. The data is already being disseminated; should the FEC provide dumps in multiple systems? Do they support every db platform? No-SQL?

In other words, "straight up SQL" is a matter of opinion and perspective. I'd rather the FEC spend its time on improving its data offerings, not duplicating what it already has.

**Comment by LindsayYoung** [comment]
duplicate of 18F/openFEC/issues/1329

**Comment by Solomon** [comment]
I think providing the database dumps is still worthwhile, and the benefit for the extra work would outweigh the effort involved.

In particular, opening up a dump of the data would allow people who have basic SQL skills to run a `pg_restore` and immediately start querying the data. There are a lot of people who know SQL, but don't have the skills to build a data pipeline.

With CSV files, even for people who have the skills to write a data pipeline, it could take a few months of part time work to get the data into postgres. It took me around 3 months of part time work to build a pipeline for the OpenSecrets data, and it's still a big pain to keep updated. This is a prohibitive amount of work for a lot of people and institutions who would want to use this data.

By creating the dump, you also open up this data to be used in hackathons, weekend projects, and a host of other places where people will be scared away by the work of creating the pipeline.

I would be very open to implementing this with as little overhead as possible, in a way that made sense to the FEC / 18f. I wouldn't expect daily releases. Even if you only released a database dump every month or every quarter, it would be incredibly useful. Even if you only released one database dump ever, people would get a lot of benefit from it.
